astropy/table/row.py

```diff
@@ -167,6 +167,14 @@ def colnames(self):
     def dtype(self):
         return self._table.dtype
 
+    def get(self, key, default=None):
+        """
+        Extract a value from the row if the key is present otherwise return the value of ``default``
+        """
+        if key in self.colnames:
```

Comment:
This probably needs to be more like your version on slack with a try/except, because `row[key]` can accept more than just column names. I.e. column number or a tuple of names work.